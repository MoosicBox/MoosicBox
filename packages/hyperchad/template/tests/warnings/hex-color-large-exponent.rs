//! Test case for hex colors with very large exponents
//! 
//! Pattern like #1e999 is valid scientific notation (1e999 is a huge number)
//! and will tokenize successfully as a float literal.
//!
//! However, it results in only 5 characters: "1e999"
//! This should fail validation because hex colors must be 3, 6, or 8 chars.
//!
//! This test documents that large exponents are valid float literals but
//! produce invalid hex color lengths.

use hyperchad_template::container;

fn main() {
    // This should fail with: "Invalid hex color '#1e999'.
    // Hex colors must be 3, 6, or 8 hexadecimal digits"
    let _ = container! {
        div color=#1e999 { "5 characters - invalid length" }
    };
}
