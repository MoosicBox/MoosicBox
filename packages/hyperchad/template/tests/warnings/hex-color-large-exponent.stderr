error: Invalid hex color '#1e999'. Hex colors must be 3, 6, or 8 hexadecimal digits (0-9, a-f)
  --> tests/warnings/hex-color-large-exponent.rs:18:19
   |
18 |         div color=#1e999 { "5 characters - invalid length" }
   |                   ^^^^^^
